python/semantic_kernel/connectors/memory/postgres/postgres_collection.py

```diff
@@ -360,38 +417,167 @@ async def _create_index(self, table_name: str, vector_field: VectorStoreRecordVe
 
         logger.info(f"Index '{index_name}' created successfully on column '{column_name}'.")
 
+    # endregion
+    # region: VectorSearchBase implementation
+
     @override
-    async def does_collection_exist(self, **kwargs: Any) -> bool:
-        """Check if the collection exists."""
+    async def _inner_search(
+        self,
+        options: VectorSearchOptions,
+        search_text: str | None = None,
+        vectorizable_text: str | None = None,
+        vector: list[float | int] | None = None,
+        **kwargs: Any,
+    ) -> KernelSearchResults[VectorSearchResult[TModel]]:
         if self.connection_pool is None:
             raise VectorStoreOperationException(
                 "Connection pool is not available, use the collection as a context manager."
             )
 
+        fields = [(field.name, field) for field in self.data_model_definition.fields.values()]
+
+        if vector is not None:
+            query, params, return_fields = self._construct_vector_query(vector, fields, options, **kwargs)
+        elif search_text:
+            raise VectorSearchExecutionException("Text search not supported.")
+        elif vectorizable_text:
+            raise VectorSearchExecutionException("Vectorizable text search not supported.")
+
         async with self.connection_pool.connection() as conn, conn.cursor() as cur:
-            await cur.execute(
-                """
-                SELECT table_name
-                FROM information_schema.tables
-                WHERE table_schema = %s AND table_name = %s
-                """,
-                (self.db_schema, self.collection_name),
+            await cur.execute(query, params)
+            rows = await cur.fetchall()
+            # Add the distance to the results
+            results = [convert_row_to_dict(row, return_fields) for row in rows]
```

Comment:
what happens when include_vectors is false with this mapping?

Reply:
Then vectors won't be included in the select list, and therefore not in the row, and it will also be excluded from the return_fields.

I did however find an issue where I was only excluding the vector field the search was being performed on; changed to exclude any vector fields if include_vectors is false.